persistence-zookeeper/src/main/scala/com/mesosphere/usi/storage/zookeeper/PodRecordRepository.scala

```diff
@@ -0,0 +1,48 @@
+package com.mesosphere.usi.storage.zookeeper
+
+import akka.util.ByteString
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+import com.mesosphere.usi.repository.{PodRecordRepository => PodRecordRepositoryInterface}
+import com.mesosphere.usi.storage.zookeeper.PersistenceStore.Node
+import org.apache.zookeeper.KeeperException.NoNodeException
+
+import scala.async.Async.{async, await}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class PodRecordRepository(val store: PersistenceStore) extends PodRecordRepositoryInterface {
+
+  override def create(record: PodRecord): Future[PodId] = async {
+    val data = record.toString
+    val node = Node(record.podId.value, ByteString(data))
+    await(store.create(node))
+    record.podId
+    // TODO: RecordAlreadyExistsException
+  }
+
+  override def read(recordId: PodId): Future[Option[PodRecord]] = async {
+    // We uncheck our match because the only failure case is the [[NoNodeException]]. The future fails in all other cases.
+    (await(store.read(recordId.value)): @unchecked) match {
```

Comment:
I'm inclined to change the storage read method. @zen-dog, what was the motivation of `Try` over `Option`?

Reply:
To distinguish between empty nodes and failed reads?

Reply:
`Failure(NoNodeException)` is a missing node. IMHO this would be a `None`.

Reply:
I might've misunderstood the question - where is the `Try` you talk about? 

To recapture:
- `Future` success/failure signals whether the node exists or not
- `Option` signals whether the node was empty.